### PR TITLE
Fix unit test for builder

### PR DESCRIPTION
**Pull request by viveksyngh**
This commit fixes issue with unit tests in builder package.

Fixes: #622

Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
#622 

## How Has This Been Tested?

Since it is only unit tests update .. No need to test the functionalities. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: builder/build_test.go [hunk 1]
```diff
@@ -287,20 +287,49 @@ func Test_buildFlagSlice(t *testing.T) {
...
+		if _, ok := outputValueMap[outputValue]; ok {
+			outputValueMap[outputValue]++
+		} else {
+			outputValueMap[outputValue] = 1
+		}
+	}
+
+	for key, expectedValue := range expectedValueMap {
+
+		if value, ok := expectedValueMap[key]; ok {
```
Comment (LucasRoesler) [inline builder/build_test.go]: Should this use the `outputValueMap` instead? 

```
		if value, ok := outputValueMap[key]; ok {
```

You iterate over the `expectedValueMap` and then select again form `expectedValueMap`, it seems tautological.


You could also reduce the amount of nesting and make the code a little more symantec by doing something like this 

```go
for key, expectedValue := range expectedValueMap {
	value, exists := outputValueMap[key]
	if !exists || value != expectedValue  {
		return false
	}
}
```
I think this if statement reads closer to how your would describe the code, "if the value does not exist or is not the expected value, return false"  The order becomes important in the if statement, because you only want to check the value after you know the value exists.


I think you need to include a length check to ensure that there are no extra values in the output map. This should probably be the first thing that happens.  

Alternatively, this can also be verified while you construct the `outputValueMap`. If you find a value that does not exist in the `expectedValueMap`, then you can return before you even check the check the values in this loop. 


Lastly, the role of the values might be clearer if we use `actual*` instead of `output*`.   This will align closer to the error messages we use in other tests.

Comment (viveksyngh) [reply builder/build_test.go]: Thanks for the feedback. Yes that should be `outputValueMap '. 

I have updated with all changes from review comments.

**Comment by zeerorg** [comment]
Yep seems like it was passing false positives with previous commit. Sorry should have tested it more thoroughly. The current commit fixes the issue.

**Comment by viveksyngh** [comment]
@zeerorg Yes we can add a different unit test function to test negative case.

**Comment by alexellis** [comment]
Derek set milestone: 0.8.10
